src/tracer.rs

```diff
@@ -98,12 +107,65 @@ impl ObjectSubclass for TracingTracerPriv {
     fn new() -> Self {
         Self {
             span_stack: thread_local::ThreadLocal::new(),
+            state: Default::default(),
         }
     }
 }
 
 impl ObjectImpl for TracingTracerPriv {
     fn constructed(&self) {
+        if let Some(params) = self.obj().property::<Option<String>>("params") {
+            let tmp = format!("params,{}", params);
+            let structure = gstreamer::Structure::from_str(&tmp).unwrap_or_else(|e| {
+                eprintln!("Invalid params string: {:?}: {e:?}", tmp);
+                gstreamer::Structure::new_empty("params")
+            });
+
+            if let Ok(gst_logs_threshold) = structure.get::<&str>("gst-logs-threshold") {
+                eprintln!(
+                    "Integrating `{gst_logs_threshold}` GStreamer logs as part of our tracing"
+                );
+
+                crate::integrate_events();
+                gstreamer::debug_remove_default_log_function();
+                gstreamer::debug_set_threshold_from_string(gst_logs_threshold, true);
+            }
+
+            if let Ok(enable_fmt) = structure.get::<bool>("enable-fmt") {
+                if enable_fmt {
+                    eprintln!("Enabling fmt tracing");
+                    tracing_subscriber::fmt::init();
+                }
+            }
+
+            if let Ok(tracing_chrome) = structure.get::<bool>("tracing-chrome").map_or_else(
+                |_| structure.get::<gstreamer::Structure>("tracing-chrome"),
+                |v| {
+                    if v {
+                        Ok(gstreamer::Structure::new_empty("tracing-chrome"))
+                    } else {
+                        Err(gstreamer::structure::GetError::FieldNotFound {
+                            name: "tracing-chrome (or wrong type)",
+                        })
+                    }
+                },
+            ) {
+                #[cfg(feature = "tracing-chrome")]
```

Comment:
I would perhaps pursue this as a separate crate (even if part of this same project,) or at least a distinct object.

While supporting just `tracing-chrome` is not a terribly high overhead, this approach seems like it would hard to scale to additional kinds of tracing outputs if people started asking for them.

(I also personally dislike crate features due to their unification semantics, which hurts use of this project as a library.)

Reply:
Would a  `chrometracing` subclass of this tracer work for you?

Reply:
Yeah, I think a subclass(es) would be alright.